src/components/Todo.tsx

```diff
@@ -0,0 +1,354 @@
+"use client";
+import React, { useState } from "react";
+import { Button } from "./ui/button";
+import type { TodoItemType } from "@/types/todo.types";
+import { ChevronDown, PlusIcon, SettingsIcon, Trash2, XIcon } from "lucide-react";
+import {
+  Dialog,
+  DialogContent,
+  DialogTitle,
+} from "@radix-ui/react-dialog";
+import { DialogFooter, DialogHeader } from "./ui/dialog";
+import { Input } from "@/components/ui/input";
+
+type TodoDashBoardProps = {
+  variant: TodoEnum;
+  todos: Array<TodoItemType>;
+  // addTodo: (todoType: TodoItemType, name: string) => void;
+  addTodo: (todoType: TodoEnum, name: string) => void;
+};
+
+export enum TodoEnum {
+  TODAY = "Today",
+  THIS_WEEK = "This Week",
+  EVENTUALLY = "Eventually",
+}
+
+// Actual Component
+const TodoHeader = () => {
+  return (
+    <div className="m-2 flex flex-wrap justify-end p-1">
+      <button className="">
+        <SettingsIcon />
+      </button>
+    </div>
+  );
+};
+
+const today = new Date();
+
+// Array of abbreviated day names
+const days = [
+  "Sunday",
+  "Monday",
+  "Tuesday",
+  "Wednesday",
+  "Thursday",
+  "Friday",
+  "Saturday",
+];
+// Array of abbreviated month names
+const months = [
+  "January",
+  "February",
+  "March",
+  "April",
+  "May",
+  "June",
+  "July",
+  "August",
+  "September",
+  "October",
+  "November",
+  "December",
+];
```

Comment:
Remove the unnecessary code like months and days, I think we dont need it, 
Also pls remove all the comments which are not needed